packages/portal-contracts/src/Portal.sol

```diff
@@ -308,13 +320,21 @@ contract Portal is Owned {
         uint128 amountSats,
         bytes20 destScriptHash
     ) public payable returns (uint256 escrowID) {
-        escrowID = orderID * 1e9;
+        escrowID = nextEscrowID++;
         Order storage o = orderbook[orderID];
         require(o.amountSats > 0, "Order already filled"); // Must be a bid
-        require(o.amountSats == int128(amountSats), "Amount incorrect");
+        require(o.amountSats >= int128(amountSats), "Amount incorrect");
+
+        uint256 totalValue = amountSats * o.priceTokPerSat;
+        int256 orderSats = int256(o.amountSats).fromInt();
+        int256 fillSats = int256(uint256(amountSats)).fromInt();
+        int256 stake = int256(o.stakedTok).fromInt();
+
+        int256 ratio = fillSats.div(orderSats);
+        uint256 portionOfStake = uint256(stake.mul(ratio)) / 1e18;
```

Comment:
we may be able to compute this without that fixed-point library, something like

```solidity
uint256 portionOfStake = o.stakedTok * uint256(amountSats) / o.amountSats;
```

To guarantee this doesn't overflow, we could enforce `o.stakedTok < 2^128` during order creation.